opensrp-app/src/main/java/org/smartregister/login/task/RemoteLoginTask.java

```diff
@@ -43,37 +59,98 @@ protected void onPreExecute() {
+                    Bundle userData = getOpenSRPContext().userService().saveUserGroup(mUsername, mPassword, loginResponse.payload());
 
-                JSONObject data = new JSONObject();
-                data.put(AllConstants.PREF_KEY.SETTINGS, settings);
-                loginResponse.setRawData(data);
+                    mAccountManager.addAccountExplicitly(account, response.getRefreshToken(), userData);
+                    mAccountManager.setAuthToken(account, mLoginView.getAuthTokenType(), response.getAccessToken());
+                    mAccountManager.setPassword(account, response.getRefreshToken());
+                    mAccountManager.setUserData(account, AccountHelper.INTENT_KEY.ACCOUNT_GROUP_ID, userData.getString(AccountHelper.INTENT_KEY.ACCOUNT_GROUP_ID));
+                    mAccountManager.setUserData(account, AccountHelper.INTENT_KEY.ACCOUNT_NAME, userData.getString(AccountHelper.INTENT_KEY.ACCOUNT_NAME));
+                    mAccountManager.setUserData(account, AccountHelper.INTENT_KEY.ACCOUNT_PASSWORD, userData.getString(AccountHelper.INTENT_KEY.ACCOUNT_PASSWORD));
+                    mAccountManager.setUserData(account, AccountHelper.INTENT_KEY.ACCOUNT_PASSWORD_SALT, userData.getString(AccountHelper.INTENT_KEY.ACCOUNT_PASSWORD_SALT));
 
-            } catch (JSONException e) {
-                Timber.e(e);
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                        mAccountManager.notifyAccountAuthenticated(account);
+                    }
+
+                    if (getOpenSRPContext().userService().getGroupId(mUsername) != null && CoreLibrary.getInstance().getSyncConfiguration().isSyncSettings()) {
+
+
+                        publishProgress(R.string.loading_client_settings);
+
+                        SyncSettingsServiceHelper syncSettingsServiceHelper = new SyncSettingsServiceHelper(getOpenSRPContext().configuration().dristhiBaseURL(), getOpenSRPContext().getHttpAgent());
+
+                        try {
+                            JSONArray settings = syncSettingsServiceHelper.pullSettingsFromServer(Utils.getFilterValue(loginResponse, CoreLibrary.getInstance().getSyncConfiguration().getSyncFilterParam()), response.getAccessToken());
+
+                            JSONObject prefSettingsData = new JSONObject();
+                            prefSettingsData.put(AllConstants.PREF_KEY.SETTINGS, settings);
+                            loginResponse.setRawData(prefSettingsData);
+
+                        } catch (JSONException e) {
+                            Timber.e(e);
+                        }
+
+                    }
+                }
+
+            } else {
+                throw new AccountsException("Could not fetch OAuth Configuration");
             }
 
+        } catch (Exception e) {
+
+            loginResponse = CUSTOM_SERVER_RESPONSE.withMessage(e.getMessage());
```

Comment:
Can we change and pass a static error, e.g Error failed during server login instead of the actual message